File: java/src/test/java/problems1_100/ContainerWithMostWaterTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ContainerWithMostWaterTest extends SampledTest {
    @Test
    public void test() {
        for (ContainerWithMostWater solution : ContainerWithMostWater.values()) {
            for (ContainerWithMostWaterSample sample
                : getSamples(ContainerWithMostWaterSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.maxArea(sample.input.height))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/ContainerWithMostWaterTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ContainerWithMostWaterTest : SampledTest() {
    @Test
    fun test() =
        ContainerWithMostWater.entries.forEach { solution ->
            getSamples<Array<ContainerWithMostWaterSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.maxArea(input.height))
                    .isEqualTo(output)
            }
        }
}
